tests/example-tbb/example_tbb.cpp

```diff
@@ -0,0 +1,39 @@
+#include "example_tbb.h"
+
+
+#include <ctime>
+#include <vector>
+#include <algorithm>
+#include <execution>
+
+using namespace std;
+
+
+void sleep(float seconds){
+    clock_t startClock = clock();
+    float secondsAhead = seconds * CLOCKS_PER_SEC;
+    // do nothing until the elapsed time has passed.
+    while(clock() < startClock+secondsAhead);
+    return;
+}
+
+bool parallel_tester(int elements) {
+    int ops{};
+    bool is_par{true};
+    std::vector<int> foo(elements);
+    std::for_each(
+        std::execution::par,
+        foo.begin(),
+        foo.end(),
+        [&](auto&& item)
+        {
+            item = ops;
+            if (ops != 0) {
+                is_par = false;
+            }
+            sleep(0.1);
+            ++ops;
+        });
```

Comment:
The way I understand this code, it'll return false as soon as not _all_ elements are processed at the same time. In combination with the test calling this with `elements == 3`, it requires at least three hardware threads for this to ever pass. This seems a bit too restrictive to me. Wouldn't it be better to check if the vector has at least two entries with the same value after the `std::for_each` has returned? That would also require _some_ level of parallelism, but would not depend on passing a number not greater than the number of hardware threads. It could look something like this:

```c++
bool parallel_tester(int elements) {
    std::atomic<int> ops{};
    std::vector<int> ops_count(elements);
    std::for_each(
        std::execution::par,
        ops_count.begin(),
        ops_count.end(),
        [&](auto&& item)
        {
            item = ops;
            sleep(0.1);
            ++ops;
        });
    for (auto i = 0UL; i < ops_count.size(); ++i) {
        for (auto j = i + 1; j < ops_count.size(); ++j) {
            if (ops_count[i] == ops_count[j]) {
                return true;
            }
        }
    }
    return false;
}
```

`parallel_tester(1)` would return `false` with the code above, so that test would need some adaptation. I'd find it reasonable to completely drop it.